docs/source/image_types.rst

```diff
@@ -105,3 +105,93 @@ Implementers of new toolkits will likely want to write their own versions of
+-----------------------------------------
+
+The :class:`~pyface.image_resource.ImageResource` system is built around
+supplying image files which are local to the place where they are being used.
+Sometimes, however, you want your images to be available throughout the
+application.  Pyface provides the :class:`~pyface.image.image.ImageLibrary`
+to allow for this use case.
+
+The :class:`~pyface.image.image.ImageLibrary` is a global object that
+holds a catalog of :class:`~pyface.image.image.ImageVolume` objects that in
+turn contain images and associated metadata.  Image volumes are either
+directories or zipfiles containing images and metadata files.
+
+Accessing Images and Image Metadata
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you give either a :class:`~pyface.ui_traits.Image` an image name which
+starts with ``@``, then it will interpret the name to be the id of an
+image in the :class:`~pyface.image.image.ImageLibrary` of the form
+``@<volume-name>:<image-name>``.  Alternatively, you can ask the image
+library directly for an image resource that corresponds to the id via the
+:meth:`~pyface.image.image.ImageLibrary.image_resource` method:
+
+.. code-block:: python
+
+    from pyface.image.image import ImageLibrary
+    from pyface.actions.api import Action
+
+    red_ball_image = ImageLibrary.image_resource("@icons:red_ball")
+    action = Action(image=red_ball_image)
+
+Each image has an :class:`~pyface.image.image.ImageInfo` metadata object
+associated with it which can be obtained via the
+:meth:`~pyface.image.image.ImageLibrary.image_info` method.  A list of all
+known image metadata can be obtained from the
+:attr:`~pyface.image.image.ImageLibrary.images` trait.
+
+:class:`~pyface.image.image.ImageVolume` Instances
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Image volumes are represented by the :class:`~pyface.image.image.ImageVolume`
+class.  Image volumes may have multiple aliases in addition to their primary
+name, and these aliases can be used as part of the id of an image.  The
+volume which stores an image can be found using the
+:meth:`~pyface.image.image.ImageLibrary.find_volume` method.  The
+:attr:`~pyface.image.image.ImageLibrary.catalog` and
+:attr:`~pyface.image.image.ImageLibrary.aliases` attributes are dictionaries
+that map names to volume objects.
+
+By default the :class:`~pyface.image.image.ImageLibrary` is initialized
+with:
+
+- an ``"application"`` volume which is assumed to be in a "library" directory
+  next to the application executable (ie. whatever is in ``sys.argv[0]``
+  which is usually the main module's path);
+- the ``std`` and ``icons`` volumes in ``pyface/image/library``
+- any path listed in the ``TRAITS_IMAGES`` environment variable which contains
+  a volume.
+
+Additional volumes can be added programatically by calling
+:meth:`~pyface.image.image.ImageLibrary.add_volume` with the path of zipfile
+or directory.  Alternatively :meth:`~pyface.image.image.ImageLibrary.add_path`
```

Comment:
```suggestion
or a directory.  Alternatively :meth:`~pyface.image.image.ImageLibrary.add_path`
```

Reply:
Not needed given the fix to line 170: "a zipfile or directory" is OK English and a bit less pedantic-sounding than "a zipfile or a directory"